operations/mimir-mixin/dashboards/writes.libsonnet

```
@@ -206,6 +235,235 @@ local filename = 'mimir-writes.json';
         )
       )
     )
+    .addRowIf(
+      $._config.show_ingest_storage_panels,
+      ($.row('Ingester (ingest storage: fetching and processing records)'))
+      .addPanel(
+        $.timeseriesPanel('Responses / sec') +
+        $.panelDescription(
+          'Responses / sec',
+          |||
+            Rate of responses from Kafka brokers. Client can return multiple responses ("fetches") at once. Some of the responses may be failures.
+          |||
+        ) +
+        $.queryPanel(
+          [
+            |||
+              sum (rate (cortex_ingest_storage_reader_fetches_total{%s}[$__rate_interval]))
+              -
+              sum (rate (cortex_ingest_storage_reader_fetch_errors_total{%s}[$__rate_interval]))
+            ||| % [$.jobMatcher($._config.job_names.ingester), $.jobMatcher($._config.job_names.ingester)],
+            'sum (rate (cortex_ingest_storage_reader_fetch_errors_total{%s}[$__rate_interval]))' % [$.jobMatcher($._config.job_names.ingester)],
+            // cortex_ingest_storage_reader_read_errors_total metric is reported by Kafka client.
+            'sum (rate (cortex_ingest_storage_reader_read_errors_total{%s}[$__rate_interval]))' % [$.jobMatcher($._config.job_names.ingester)],
+          ],
+          [
+            'fetches',
+            'failed',
+            'read errors',
+          ],
+        ) + $.aliasColors({ failed: '#FF0000', 'read errors': '#FF0000' }) + $.stack,
+      )
+      .addPanel(
+        $.timeseriesPanel('Records per fetch') +
+        $.panelDescription(
+          'Records per fetch',
+          |||
+            Number of fetched records per fetch operation.
+          |||
+        ) +
+        $.queryPanel(
+          [
+            |||
+              sum(rate(cortex_ingest_storage_reader_records_per_fetch_sum{%s}[$__rate_interval]))
+              /
+              sum(rate(cortex_ingest_storage_reader_records_per_fetch_count{%s}[$__rate_interval]))
+            ||| % [$.jobMatcher($._config.job_names.ingester), $.jobMatcher($._config.job_names.ingester)],
+            'histogram_quantile(0.99, sum by(le) (rate(cortex_ingest_storage_reader_records_per_fetch_bucket{%s}[$__rate_interval])))' % [$.jobMatcher($._config.job_names.ingester)],
+          ],
+          [
+            'Average',
+            '99th percentile',
+          ],
+        ),
+      )
+      .addPanel(
+        $.timeseriesPanel('Processing Latency') +
```

If we add "Kafka records / sec" then may be clearer name this "Kafka records processing latency" to clarify what we're processing

```suggestion
        $.timeseriesPanel('Kafka records processing latency') +
```